doc/exts/pylint_messages.py

```diff
@@ -0,0 +1,208 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+"""Script used to generate the messages files."""
+
+import os
+from collections import defaultdict
+from typing import DefaultDict, Dict, List, NamedTuple, Optional, Tuple
+
+from sphinx.application import Sphinx
+
+from pylint.checkers import initialize as initialize_checkers
+from pylint.extensions import initialize as initialize_extensions
+from pylint.lint import PyLinter
+from pylint.message import MessageDefinition
+from pylint.utils import get_rst_title
+
+# PACKAGE/docs/exts/pylint_extensions.py --> PACKAGE/
+PYLINT_BASE_PATH = os.path.dirname(
+    os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+)
+"""Base path to the project folder"""
+
+
+MSG_TYPES = {
+    "I": "information",
+    "C": "convention",
+    "R": "refactor",
+    "W": "warning",
+    "E": "error",
+    "F": "fatal",
+}
+
+
+class MessageData(NamedTuple):
+    checker: str
+    id: str
+    name: str
+    definition: MessageDefinition
+
+
+MessagesDict = Dict[str, List[MessageData]]
+OldMessagesDict = Dict[str, DefaultDict[Tuple[str, str], List[Tuple[str, str]]]]
+"""DefaultDict is indexed by tuples of (old name symbol, old name id) and values are
+tuples of (new name symbol, new name category)
+"""
+
+
+def _register_all_checkers_and_extensions(linter: PyLinter) -> None:
+    """Registers all checkers and extensions found in the default folders."""
+    initialize_checkers(linter)
+    initialize_extensions(linter)
+
+
+def _get_all_messages(
+    linter: PyLinter,
+) -> Tuple[MessagesDict, OldMessagesDict]:
+    """Get all messages registered to a linter and return a dictionary indexed by message
```

Comment:
We should use the ``MessageDefinitionStore`` or the ``MessageIdStore``, if we need more information about the checker or something else we should add the information in MessageDefinitionStore directly, not in a helper function. (ie : ``for message in linter.msgs_store.messages`` should work)

Reply:
https://github.com/PyCQA/pylint/pull/5396/commits/f0414107e6e76f9b2861d0bc9a88417c778419b1

Reply:
There is a protected ``self._msgs_by_category`` in the ``MessageIdStore``, by making it public and maybe handling old names, we can remove most of the helper code in this file.

Reply:
I'm not sure if we should do that (as then plugin creators can use it as well and we need to support it). A little extra helper code doesn't hurt right? This is only run once every CI and every push to main. It doesn't take too long.

Reply:
Fair enough.